src/main/java/slave/Pair.java

```diff
@@ -0,0 +1,37 @@
+package slave;
+
+/**
+ * This class contains a pair of values of any types T and U
+ *
+ * @param <T>
+ * @param <U>
+ */
+public class Pair<T, U> {
+    private T t;
+    private U u;
+
+    /**
+     * creates a pair containing values t of type T and u of type U
```

Comment:
Nitpicking here but "Creates" just keeps the Javadocs more consistent with the rest of your naming.